### PR TITLE
fix: Fixed repeated auths prompt (close #652)

### LINKED ISSUE #652: On MacOS, it will ask for username/password every time after VSCode restart
- VSCode Version: 1.37.0
- OS Version: Darwin x64 18.7.0
- Extension Version:1.54.4
- System Language:English
- SVN Version:1.8.13

## Issue ##
Recently, every time when I restart VSCode, SVN will ask for username and password. However, when I offer the username and password, it works, but when I restart VSCode, it ask again. I'm nearly crazy.

## Screenshots ##
QQ20190812-233357@2x

### PR DESCRIPTION
**Pull request by edgardmessias**
Instead of find a way to save auth, use keytar as an alternative.

See: https://code.visualstudio.com/api/advanced-topics/remote-extensions#persisting-secrets

### CODE REVIEW

**Comment by JohnstonCode** [comment]
fixes #188 ?

**Comment by edgardmessias** [comment]
I no know, we need be a test, because `keytar` depends of system libs.

Need test with and without `gnome-keyring`

File: src/repository.ts [hunk 1]
```diff
@@ -919,6 +922,39 @@ export class Repository implements IRemoteRepository {
...
+    let key = "vscode.svn-scm";
+
+    const info = this.repository.info;
+
+    if (info.repository && info.repository.root) {
+      key += ":" + info.repository.root;
+    } else if (info.url) {
+      key += ":" + info.url;
+    }
+
```
Comment (JohnstonCode) [inline src/repository.ts]: could this be
```
key += ":" + info?.repository?.root ?? info.url;
```

**Comment by JohnstonCode** [comment]
Let me know if you're happy to merge

**Comment by edgardmessias** [comment]
I'm happy, lol

**Pull request merged**